files/ansible/tasks/put_sftp.yml

```diff
@@ -0,0 +1,5 @@
+---
+- name: Copy results to fileserver
+  shell: (echo "mkdir {{ sftp_dirname }}" && echo "cd {{ sftp_dirname }}" && echo "put {{ item.path }}" && echo "exit") | sftp -o ForwardAgent=yes -o ConnectTimeout=60 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no  {{sftp_user}}@{{ sftp_server_ip }}:/{{ sftp_path }}
```

Comment:
If any of those variables doesn't exist or are empty, will this fail or just proceed with an empty substitution?

Reply:
[According to this, "The default behavior from ansible and ansible.cfg is to fail if variables are undefined"](https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#forcing-variables-to-be-defined), that said I appended a "mandatory" filter to make it explicit.